src/modal/modal-stack.ts

```diff
@@ -20,75 +22,110 @@ import {NgbActiveModal, NgbModalRef} from './modal-ref';
 export class NgbModalStack {
   private _backdropFactory: ComponentFactory<NgbModalBackdrop>;
   private _windowFactory: ComponentFactory<NgbModalWindow>;
+  private _document: any;
+  private _windowAttributes = ['backdrop', 'keyboard', 'size', 'windowClass'];
 
   constructor(
-      private _applicationRef: ApplicationRef, private _injector: Injector,
+      @Inject(DOCUMENT) document, private _applicationRef: ApplicationRef, private _injector: Injector,
       private _componentFactoryResolver: ComponentFactoryResolver) {
-    this._backdropFactory = _componentFactoryResolver.resolveComponentFactory(NgbModalBackdrop);
-    this._windowFactory = _componentFactoryResolver.resolveComponentFactory(NgbModalWindow);
+    this._document = document;
   }
 
-  open(moduleCFR: ComponentFactoryResolver, contentInjector: Injector, content: any, options): NgbModalRef {
-    const containerSelector = options.container || 'body';
-    const containerEl = document.querySelector(containerSelector);
+  open(content: any, options): NgbModalRef {
+    this._buildFactories();
+    const containerEl =
+        isDefined(options.container) ? this._document.querySelector(options.container) : this._document.body;
 
     if (!containerEl) {
-      throw new Error(`The specified modal container "${containerSelector}" was not found in the DOM.`);
+      throw new Error(`The specified modal container "${options.container}" was not found in the DOM.`);
```

Comment:
This will have a broken error message if a user doesn't specify the `container` option

Reply:
If a user doesn't specify the `container` option then will use `this._document.body` and not a query selector.
`this._document.body` will be always defined, right?

Reply:
> this._document.body will be always defined, right?

In theory not always, as it can potentially be `null`: https://developer.mozilla.org/en-US/docs/Web/API/Document/body

In practice it should be always defined, but it is better to be safe than sorry...